CONTRIBUTING.md

```diff
@@ -44,6 +44,15 @@ Given that you have a compatible environment as stated above you can now set up
 +  Open the project in Android Studio
 + If you are a Skyscanner employee, search the internal documentation for _"Guide – Setup Internal Backpack Android Builds"_ and follow the instructions.
 
+## Creating components
+
+To create a new component in Compose you can use the following script to setup the most common required parts of a component:
+
+```
+python3 scripts/generate-component.py NameOfComponent
```

Comment:
I think we need to do this as a Gradle task. Having 3rd building framework (in addition to Gradle and Node) seems flaky. Besides, I believe Python needs its own environment (sooner or later), and we want the contribution and setup to be as easy as possible.

Reply:
I originally started off writing this using our current token generation system and got something working, but it's not ideal: it would include all the `public` / `Unit` modifiers/return values, so you'd either have this polluted in the code base or need to remove it manually. kotlinpoets is for generated files that don't need to be touched manually.

The problem with gradle is it's not really built for things like this - it's a build system. So even just providing an input to it is not as simple as in python or other scripting languages (both in development terms & usage terms).

Output using a gradle task also isn't that obvious (as you know from the ktlint output), so it'll be harder to use/figure out what manual steps to take and what actually happened in the script.

I also think this script needs to be extended to take step by step inputs from users (with defaults) & confirmation steps. For example to make it clear what format the component name should be in, giving users options to change the default package name/component name (for example for something like fab where you might want to use a shorter name in places). That isn't possible with a gradle task either.

So I definitely understand where you're coming from, but also we need to use the right tool for the right task and Gradle isn't the right tool for this.